### PR TITLE
CODE-90 config eslint and move prettier to eslint rules

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
https://osi-cwds.atlassian.net/browse/CODE-90

This is the begin to configuring the eslint for design-system. Currently configured with standard prettier rules for the libs we export out. The follow up PR's will be addressing the eslint standard, jsx-a11y and react.

### CODE REVIEW

**Comment by eckdanny-osi** [comment]
The last holdout for us moving the workspace root devDep `babel6` to `babel7` was `storybook`, but the 4.x branch has been stabilizing since the thanksgiving release. Does eslint introduce a new dep on `babel6`?

File: libs/core/js/core.js [hunk 1]
```diff
@@ -30,29 +30,29 @@ DS.colors = {
...
-  '100':        Color.hsl(0, 0, 96).hex(),   // ($ray)
-  '200':        Color.hsl(0, 0, 91).hex(),   // ($grouper)
-  '300':        Color.hsl(0, 0, 82).hex(),   // ($trout)
-  '400':        Color.hsl(0, 0, 60).hex(),   //
-  '500':        Color.hsl(0, 0, 50).hex(),   //
-  '600':        Color.hsl(0, 0, 40).hex(),   //
-  '700':        Color.hsl(0, 0, 36).hex(),   // ($wahoo)
-  '800':        Color.hsl(0, 0, 20).hex(),   //
-  '900':        Color.hsl(0, 0, 10).hex(),   // ($shark)
+  '100': Color.hsl(0, 0, 96).hex(), // ($ray)
```
[the remark this answers is not in the document]
Comment (NagaPradeepGajula-osi) [reply libs/core/js/core.js]: I will configure eslint to ignore the files specified in prettier.

File: package.json [hunk 2]
```diff
@@ -20,7 +20,9 @@
     "build:watch": "lerna run --scope @cwds/* --parallel build:watch",
     "dev": "yarn build:watch",
     "storybook": "start-storybook -p 9001 -c .storybook",
-    "release": "lerna publish --conventional-commits --force-publish"
+    "release": "lerna publish --conventional-commits --force-publish",
+    "lint": "eslint --ext .jsx --ext .js libs/",
+    "fix-lint": "eslint --ext .jsx --ext .js --fix libs/;exit 0;"
```
[the remark this answers is not in the document]
Comment (NagaPradeepGajula-osi) [reply package.json]: We don't need it. Should be cleaned up.

File: package.json [hunk 3]
```diff
@@ -39,6 +41,7 @@
     "@storybook/addon-knobs": "^3.4.11",
     "@storybook/react": "^3.4.11",
     "babel-cli": "^6.26.0",
+    "babel-eslint": "10.0.1",
```
Comment (eckdanny-osi) [inline package.json]: did the `yarn.lock` get of date?

`yarn install --frozen-lockfile` fails

File: package.json [hunk 4]
```diff
@@ -39,6 +41,7 @@
     "@storybook/addon-knobs": "^3.4.11",
     "@storybook/react": "^3.4.11",
     "babel-cli": "^6.26.0",
+    "babel-eslint": "^8.2.5",
```
[the remark this answers is not in the document]
Comment (NagaPradeepGajula-osi) [reply package.json]: done

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (NagaPradeepGajula-osi) [reply package.json]: done

File: package.json [hunk 5]
```diff
@@ -20,7 +20,9 @@
     "build:watch": "lerna run --scope @cwds/* --parallel build:watch",
     "dev": "yarn build:watch",
     "storybook": "start-storybook -p 9001 -c .storybook",
-    "release": "lerna publish --conventional-commits --force-publish"
+    "release": "lerna publish --conventional-commits --force-publish",
+    "lint": "eslint --ext .jsx --ext .js libs/",
+    "fix-lint": "eslint --ext .jsx --ext .js --fix libs/"
```
Comment (eckdanny-osi) [inline package.json]: can we follow the `general:specific:micro` convention (piggybacking add'l args when possible):

```suggestion
    "lint:fix": "yarn lint --fix"
```

File: package.json (hunk 3 above)
Comment (NagaPradeepGajula-osi) [reply package.json]: argh! me being dumb didn't do yarn install after updating JSON. This is what happens if we work while in a meeting.

File: package.json (hunk 5 above)
Comment (NagaPradeepGajula-osi) [reply package.json]: 👍

**Comment by eckdanny-osi** [comment]
my editor's workspace settings needed a small change to keep prettier[+eslint] integration.

Should we track workspace config/cruft now that there are many contributors?

File: package.json (hunk 3 above)
Comment (eckdanny-osi) [reply package.json]: 💁🎸🎤🎄

**Comment by NagaPradeepGajula-osi** [comment]
Yeah, I turned it off my "editor.formatOnSave": true, to false in workspace settings to fix the whiteSpaces. I can make this change.
and as for the auto save with prettier and eslint config I can configure eslint.autosave.

**Comment by NagaPradeepGajula-osi** [comment]
done

**Comment by NagaPradeepGajula-osi** [comment]
@eckdanny-osi is it a green to turn off editor.formatonsave to false in workspace settings?

**Comment by eckdanny-osi** [comment]
~we currently **are not**~ we **do** track ide cruft in vcs (wow, had a special moment myself there)

appending

```
"prettier.eslintIntegration": true
```

should suffice i think

**Comment by NagaPradeepGajula-osi** [comment]
@eckdanny-osi github says I am not authorized to merge this PR. Any idea?

**Pull request merged**